_posts/2020-12-17-solidity-v0.8.0-release-announcement.md

```diff
@@ -0,0 +1,237 @@
+
+These are only some of the more important breaking changes in 0.8.0., please read the changelog below for the full list!
+
+### Checked Arithmetic
+
+The "Checked Arithmetic" feature of Solidity 0.8.x consists of three sub-features:
+- Revert on assertion failures and similar conditions instead of using the invalid opcode.
+- Actual checked arithmetic.
+- ``unchecked`` blocks.
+
+#### Revert on assertion failures and similar conditions instead of using the invalid opcode
+
+Prevously, internal errors like division by zero, failing assertions, array access out of bounds,
+etc., would result in an invalid opcode being executed. The idea was to distinguish
+these more critical errors from non-critical errors that lie outside of the domain
+of the Smart Contract programmer like invalid input, not enough balance for a token transfer,
+and so on. These non-critical errors use the ``revert`` opcode and optionally add an error description.
+
+The problem with the invalid opcode is that, in contrast to the ``revert`` opcode, it consumes all gas available.
+This makes it very expensive and you should try to avoid it at all cost.
+
+We wanted to consider arithmetic overflow as a critical error, but did not want to cause it to consume all gas.
+As a middle ground, we chose to use the ``revert`` opcode, but provide a different error data
+so that static (and dynamic) analysis tools can easily distinguish them.
+
+More specifically:
+ - Non-critical errors will revert either with empty error data or with error data that
+   corresponds to the ABI-encoding of a function call to a function with the signature ``Error(string)``.
+ - Critical errors revert with error data that corresponds to the ABI-encoding of a function call
+   to a function with the signature ``Panic(uint256)``.
+
+Because of that, we also use the term "panic" for such critical errors.
+
+To distinguish the two situations, you can take a look at the first four bytes of the return data in case
+of a failure. If it is ``0x4e487b71``, then it is a panic, if it is ``0x08c379a0``
+or if the message is empty, it is a "regular" error.
+
+Panics use specific error codes to distinguish certain situations in which the panic is triggered.
+This is the current list of error codes:
+
+1. 0x01: If you call ``assert`` with an argument that evaluates to false.
+2. 0x11: If an arithmetic operation results in underflow or overflow outside of an ``unchecked { ... }`` block.
+3. 0x12: If you divide or modulo by zero (e.g. ``5 / 0`` or ``23 % 0``).
+4. 0x21: If you convert a value that is too big or negative into an enum type.
+5. 0x31: If you call ``.pop()`` on an empty array.
+6. 0x32: If you access an array, ``bytesN`` or an array slice at an out-of-bounds or negative index (i.e. ``x[i]`` where ``i >= x.length`` or ``i < 0``).
+7. 0x41: If you allocate too much memory or create an array that is too large.
+8. 0x51: If you call a zero-initialized variable of internal function type.
```

Comment:
The list is not the same as [the one we have in the docs](https://docs.soliditylang.org/en/breaking/control-structures.html#panic-via-assert-and-error-via-require).

`0x22` is missing:
```markdown
5. 0x22: If you access a storage byte array that is incorrectly encoded.
```